### PR TITLE
[dict.cc] feat: Translate from Clipboard command

### PR DESCRIPTION
**Pull request by sjdonado**
## Description

Introducing "Translate from Clipboard" command. Past features untouched, only the folder structure refactored in order to follow DRY.

## Screencast

https://user-images.githubusercontent.com/27580836/219883277-58267bb1-54fd-4737-b693-0d0a61ae785c.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

The extensions suffer from [Empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/dict-cc/src/components/TranslationsList.tsx [hunk 1]
```diff
@@ -50,14 +42,28 @@ export default function Command() {
...
+    if (isSearchFromClipboard) {
+      (async () => {
+        const clipboardText = await Clipboard.readText();
+
+        if (clipboardText && clipboardText !== searchText) {
+          fetchTranslations(clipboardText);
+        }
+      })();
+    }
+  }, []);
```
Comment (natterstefan) [inline extensions/dict-cc/src/components/TranslationsList.tsx:59]:
```suggestion
  }, [isSearchFromClipboard , fetchTranslations, searchText]);
```

PS: @pernielsentikaer, consider adding this ESLint rule to the extension template: https://www.npmjs.com/package/eslint-plugin-react-hooks.

**Review by natterstefan — COMMENTED** [review]
Hi @sjdonado,

thanks again for adding one more feature to the extension. I appreciate it. I've added some comments we can discuss.

File: extensions/dict-cc/src/components/TranslationsList.tsx [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
+export function TranslationsList({ isSearchFromClipboard }: ITranslationsListProps) {
   const [translations, setTranslations] = useState<Translations[] | undefined>();
   const [url, setUrl] = useState<string | undefined>();
   const [languages, setLanguages] = useState<[/* source */ Languages, /* target */ Languages] | undefined>();
   const [loading, setLoading] = useState(false);
+
   const [searchText, setSearchText] = useState("");
 
-  const onSearchTextChange = async (searchTerm: string) => {
+  const fetchTranslations = async (searchTerm: string) => {
```
[the remark this answers is not in the document]
Comment (sjdonado) [reply extensions/dict-cc/src/components/TranslationsList.tsx]: Good catch! applied :)

File: extensions/dict-cc/src/components/TranslationsList.tsx (hunk 1 above)
Comment (pernielsentikaer) [reply extensions/dict-cc/src/components/TranslationsList.tsx]: CC @thomaslombart

**Comment by sjdonado** [comment]
Hi @pernielsentikaer ready for a second round :)

**Review by natterstefan — APPROVED** [review]
Looks good to me. 🦊

(PS: I did not test it with Raycast directly, only reviewed the code.)

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋

Tested with success, let's go! 🧑‍🚒

**Pull request merged**